llvmlite/tests/test_binding.py

```diff
@@ -491,11 +494,12 @@ def test_parse_bitcode_error(self):
         self.assertIn("Invalid bitcode signature", str(cm.exception))
 
     def test_bitcode_roundtrip(self):
-        bc = self.module().as_bitcode()
+        bc = self.module(asm=asm_mul).as_bitcode()
```

Comment:
Why does this test has to be changed?

Reply:
Since we reuse the context for the changes, the types get a numerical suffix each time a module is parsed (i.e. `struct.somestruct.1` etc). This fails the tests since the produced IR is slightly different, even though it's sematically equivalent. `asm_mul` doesn't declare any types, so it works fine.

Another solution would be to allow the user to instantiate custom llvm contexts instead of relying on the global one. Since it's the preferred method in the newer LLVM version, it might be the better solution in the mid-term, but would be outside of the scope of this PR.

Reply:
I see.  And yes, we need to move to the new way of llvm-context at some point.